extensions/libraries/redcore/database/sqlparser/sqltranslation.php

```diff
@@ -158,16 +158,26 @@ public static function buildTranslationQuery($sql = '', $prefix = '')
 			}
 		}
 
+		// If we have a SELECT in the query then we check the reset of the params
+		$validSelect = (!empty($sql) && stristr(mb_strtolower($sql), 'select'));
+
+		// If the language is the default, there is no reason to translate
+		$isDefaultLanguage = (RTranslationHelper::getSiteLanguage() == $selectedLanguage);
+
+		// If this is the admin, but no a SOAP request we shouldn't translate
+		$app = JFactory::getApplication();
+		$isSoapApi = (strtolower($app->input->getString('api')) == 'soap');
```

Comment:
maybe this should be put into a helper function in SOAP helper, so it can be called from different places (like detecting which API is currently running)